internal/log/authorize.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import "strings"
+
+// An AuthorizeLogField is a field in the authorize logs.
+type AuthorizeLogField string
+
+// known authorize log fields
+const (
+	AuthorizeLogFieldCheckRequestID       AuthorizeLogField = "check-request-id"
+	AuthorizeLogFieldEmail                AuthorizeLogField = "email"
+	AuthorizeLogFieldHeaders              AuthorizeLogField = "headers"
+	AuthorizeLogFieldHost                 AuthorizeLogField = "host"
+	AuthorizeLogFieldImpersonateEmail     AuthorizeLogField = "impersonate-email"
+	AuthorizeLogFieldImpersonateSessionID AuthorizeLogField = "impersonate-session-id"
+	AuthorizeLogFieldImpersonateUserID    AuthorizeLogField = "impersonate-user-id"
+	AuthorizeLogFieldIP                   AuthorizeLogField = "ip"
+	AuthorizeLogFieldMethod               AuthorizeLogField = "method"
+	AuthorizeLogFieldPath                 AuthorizeLogField = "path"
+	AuthorizeLogFieldQuery                AuthorizeLogField = "query"
+	AuthorizeLogFieldRequestID            AuthorizeLogField = "request-id"
+	AuthorizeLogFieldServiceAccountID     AuthorizeLogField = "service-account-id"
+	AuthorizeLogFieldSessionID            AuthorizeLogField = "session-id"
+	AuthorizeLogFieldUser                 AuthorizeLogField = "user"
+)
+
+// DefaultAuthorizeLogFields are the default authorize log fields to log.
+var DefaultAuthorizeLogFields = []AuthorizeLogField{
+	AuthorizeLogFieldRequestID,
+	AuthorizeLogFieldCheckRequestID,
+	AuthorizeLogFieldMethod,
+	AuthorizeLogFieldPath,
+	AuthorizeLogFieldHost,
+	AuthorizeLogFieldQuery,
+	AuthorizeLogFieldIP,
+	AuthorizeLogFieldSessionID,
+	AuthorizeLogFieldImpersonateSessionID,
+	AuthorizeLogFieldImpersonateUserID,
+	AuthorizeLogFieldImpersonateEmail,
+	AuthorizeLogFieldServiceAccountID,
+	AuthorizeLogFieldUser,
+	AuthorizeLogFieldEmail,
+	AuthorizeLogFieldHeaders.ForDebugOnly(),
```

Comment:
My first impression is that `ForDebugOnly()` / `IsForDebugOnly()` mechanism feels a little confusing.

What would you think about making `DefaultAuthorizeLogFields` into a function instead? Something like this:

```
func DefaultAuthorizeLogFields(logLevel zerolog.Level) []AuthorizeLogField {
    fields := []AuthorizeLogField{
        AuthorizeLogFieldRequestID,
        AuthorizeLogFieldCheckRequestID,
        AuthorizeLogFieldMethod,
        AuthorizeLogFieldPath,
        AuthorizeLogFieldHost,
        AuthorizeLogFieldQuery,
        AuthorizeLogFieldIP,
        AuthorizeLogFieldSessionID,
        AuthorizeLogFieldImpersonateSessionID,
        AuthorizeLogFieldImpersonateUserID,
        AuthorizeLogFieldImpersonateEmail,
        AuthorizeLogFieldServiceAccountID,
        AuthorizeLogFieldUser,
        AuthorizeLogFieldEmail,
    }
    if logLevel <= zerolog.DebugLevel {
        fields = append(fields, AuthorizeLogFieldHeaders)
    }
    return fields
} 
```

(definitely optional, please let me know if it feels simpler to keep it the way you have it here)

Reply:
but then you'd have no way of defining this functionality yourself... maybe that's ok because if you want to customize it you probably don't care about log levels?